### ISSUE TITLE
User account corrupts - 2.0.0-RC[2345]

### ISSUE THREAD

**Issue by jmhunter** [nextcloud/android#1469]
### Actual behaviour
- Set up new installation of Nextcloud client on phone
- Connect to nextcloud server with username and password
- User account as shown in app changes from full name & username, to GUID & GUID
- App crashes

### Expected behaviour
- User account remains as showing username, instead of GUID
- No app crashes
 
### Steps to reproduce

This has been totally reproducible for me, with every 2.0.0 RC so far.

1.4.2 from F-Droid works fine.

1. Fresh install of application (RC4) after uninstalling previous in order to clear data
2. First-run wizard comes up; I successfully provide hostname, username and password and log in to my Nextcloud server (version 10.0 if that makes a difference)
3. Look at list of accounts, it shows up with two lines (top line: Firstname Lastname, bottom line: username@owncloudhostname) and my photo, as I’d expect - so far, all good. Clicking the drop-down here, and choosing "Manage Accounts" shows me the same detail at this point.
5. I go and do something else on my phone. Could be anything -I even tried just clicking on a different activity in the NextCloud client (e.g. "Activities")
6. When I come back into nextcloud (I have literally not done anything else in the nextcloud app yet) I look at the list of accounts in the app. Both my full name and my username have been replaced with my GUID and my photo is gone
6. I go to “manage accounts”, and do not click on any accounts.
7. In the list, my one account (with GUID now) is shown
8. I wait a short while. Nextcloud then crashes (“Nextcloud has stopped / Open App Again / Send Feedback”

### Environment data
Android version: 7.0

Device model: Samsung S7 Edge

Stock or customized system: Stock

Nextcloud app version: 2.0.0RC[2345]

Nextcloud server version: 10.0

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by mario** [comment]
Is it possible that somewhere we need guid, and somewhere username? *shrug* cc @AndyScherzinger @tobiasKaminsky

**Comment by jmhunter** [comment]
(Happy to set up a test account for a dev on my NC 10.0 server if that's easiest, btw!!)

**Comment by tobiasKaminsky** [comment]
@jmhunter would be great if you send a test account to andoid at nextcloud dot com

**Comment by jmhunter** [comment]
Thanks @tobiasKaminsky, I will do. I'm away for two days with limited connectivity - can it wait till after then? Otherwise, I can find a wifi hotspot and set an account up.. I don't want to hold up any releases..

**Comment by AndyScherzinger** [comment]
@jmhunter no worries, Tobias and I won't be available this weekend :)

**Comment by jmhunter** [comment]
Test account created and thankfully I reproduced the problem straight away using it.. that's a good sign at least.

Login details have been emailed to android@ - let me know if they don't arrive for some reason.

I also did check the nextcloud.log file on the server and as expected there are many `Login failed: 'XXXXXX-XXXX-XXXX...'` messages on the server side, once the Android client switches to displaying the GUID.
Also a matching error from LDAP - I don't think that has any bearing on the bug though, given I think the bug is client-side: `"app":"user_ldap","message":"Exception: {\"Exception\":\"OCA\\\\User_LDAP\\\\Exceptions\\\\NotOnLDAP\",\"Message\":\"No user available for the given login name on ldaps:\\\/\`

**Comment by tobiasKaminsky** [comment]
I saw that you are running a NC10 version, which is end of life.
As it might be that this error was fixed in newer releases, please upgrade before we spent more effort in debugging this.

**Comment by jmhunter** [comment]
Are we saying that the NC Android client 2.0 will require a minimum version of NC >10? That'd be disappointing, as the bug does not exist in NC Android client 1.4.3, it came in when I tried the beta client (2.0.0RC2 I think).

Upgrading the server is on my list of things to do, certainly - but I believe this to be a client-side bug rather than a server-side bug? I haven"t changed the server, yet a newer version of the client now doesn't work.

**Comment by tobiasKaminsky** [comment]
We upgraded the login mechanism to harden security and tested it on all supported (NC >=11) versions.
There will be a fallback mechanism during new login method if you wait long enough (~45s), which might work.

**Comment by jmhunter** [comment]
Thanks Tobias. Unfortunately the fallback mechanism consistently doesn't work for me - the current builds of the Android client just don't work with [my] NC10 at all.

**Comment by mario** [comment]
Can I get test account on mario@nextcloud.com please?

**Comment by Riccardo2017** [comment]
I have switched to NC 2 RC5 this week and have the same issue. I have had NC server version 12.0.0 before and yesterday I updated to 12.0.2. The same behaviour.

I can login into my server sometimes. But I can't synchronize or even see any data. My client doesn't crash, but I only can see "loading data" and nothing happens. I also see in my NC server the messages "no user available".

Before I used the NC client 1.4.3 and it worked fine.

I also get sometimes the possibility to use the old login method, but also doesn't work.

I use an Android 6.x phone. If you need, I can provide you a user account and send you also the log files.

**Comment by mario** [comment]
@Riccardo2017 thanks for the offer. This has been fixed and will be in the next RC.

**Comment by jmhunter** [comment]
I can confirm that this is fixed in 2.0.0-RC6 - thank you all!
